src/msi/msi_token.go

```diff
@@ -0,0 +1,89 @@
+package msi
+
+import (
+  "fmt"
+  "github.com/Sirupsen/logrus"
+  "io/ioutil"
+  "net/http"
+  "net/url"
+  "encoding/json"
+)
+
+type tokenJson struct {
+  AccessToken string `json:"access_token"`
+  RefreshToken string `json:"refresh_token"`
+  ExpiresIn string `json:"expires_in"`
+  ExpiresOn string `json:"expires_on"`
+  NotBefore string `json:"not_before"`
+  Resource string `json:"resource"`
+  TokenType string `json:"token_type"`
+}
+
+var (
+  log = logrus.WithField("prefix", "msi")
+)
+
+
+func GetToken(resource string, msiIdentity string) (string, error) {
+
+  logger := log.WithFields(logrus.Fields{
+    "resource":       resource,
+    "msiIdentity":     msiIdentity,
+  })
+
+  logger.Info("Got request for msi token")
+    // Create HTTP request for MSI token to access Azure Resource Manager
+    var msi_endpoint *url.URL
+    msi_endpoint, err := url.Parse("http://localhost:50342/oauth2/token")
```

Comment:
The identity service is hosted locally (i.e. `localhost`) and not in the hypervisor (i.e. `169.254.x.x`)?

Reply:
Microsoft had mentioned that in the "first drop" the MSI is hosted on localhost. They will fix that to 169.254.x.. in "Second drop".